Sources/OpenAPIRuntime/URICoder/Serialization/URISerializer.swift

```diff
@@ -180,6 +181,9 @@ extension URISerializer {
         case (.simple, _):
             keyAndValueSeparator = nil
             pairSeparator = ","
+        case (.deepObject, _):
```

Comment:
By my reading of the OpenAPI 3.1 specification, deepObject _must_ have a JSON object at the top, so an array would not be supported? https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#style-examples

Reply:
This should throw an error - deepObject doesn't support arrays.